planet/src/main/webapp/js/placesItinerary.js

```diff
@@ -13,55 +13,104 @@
 // limitations under the License.
 
 // This file contains functions related to the places part of the itinerary page
+
+import TimeRange from './TimeRange.js';
+
+// Declare global functions.
 window.openAddPlaceForm = openAddPlaceForm;
 window.closeAddPlaceForm = closeAddPlaceForm;
 
+// Declare global variables/constants.
 const database = firebase.database();
+let map = new google.maps.Map(document.getElementById('empty-map'));
 
 export function renderPlaces() {
     const userId = firebase.auth().currentUser.uid;
     const placesRef = database.ref('users/' + userId + '/places');
     placesRef.once('value', (placesSnapshot) => {
         const placesContainer = document.getElementById('places');
         placesContainer.innerHTML = '';
-        placesSnapshot.forEach((childPlace) => {
-            let placeDetailsRef = database.ref('places/' + childPlace.val());
-            placeDetailsRef.once('value', (placeDetailsSnapshot) => {
-                let placeObject = placeDetailsSnapshot.val();
-                let placeElement = createPlaceElement (placeDetailsSnapshot.key,
-                                                placeObject.name, 
-                                                placeObject.address);
-                placesContainer.appendChild(placeElement);
-            });
-        });
+        const savedPlaces = placesSnapshot.val();
+        for (const placeId in savedPlaces) {
+            // Make request with fields
+            let placeRequest = {
+                placeId: placeId,
+                fields: ['place_id','name','rating','formatted_address',
+                    'opening_hours','photos','url']
+            };
+            let service = new google.maps.places.PlacesService(map);
+            service.getDetails(placeRequest, renderPlaceDetailsCallback); 
+        }
     });
 }
 
-function createPlaceElement(ref, name, address) {
+function renderPlaceDetailsCallback(place, status) {
+    if (status == google.maps.places.PlacesServiceStatus.OK) {
+        const placesContainer = document.getElementById('places');
+        const placeElement = createPlaceElement(place);
+        placesContainer.appendChild(placeElement);
```

Comment:
do these need to be rendered in a particular order?

Reply:
It doesn't have to be any particular order. I think this function has to modified (in my next PR) because google maps javascript api can't make more than 10 requests at a time. We're not sure the best way to solve it however.
![image](https://user-images.githubusercontent.com/46226072/88310578-d6b13d80-ccdd-11ea-8173-0580f03604dd.png)
https://stackoverflow.com/questions/41946356/google-places-api-batch-requests-javascript 

Option 1 is to have a load more button that displays 10 more cards every time its clicked. The cons are it's hard to keep track of what has been loaded and what hasn't. And since the bucket is refilled 1 second at a time that means users can click "load more" twice in 10 seconds

Option 2 is to go back to saving place details in the database, but somehow update the entire database once in a while (once a month etc.). Not sure how to implement it though

Option 3 is saving the only the name a address in the database (since they likely won't change), and only get place details when the user adds the place as an event. The con is that the user won't be able to see place details (ratings, opening hours...) on the itinerary page, and they could have been useful information for users to decide which places to add and which places not to.

What do you guys think @emilyzhaoo @nabil-dbz @EkkiD @StarIU072

Reply:
I would focus on Option 1. One idea would be to keep track with a setTimeout() counter that indicates the number of requests you can make. If the user clicks "load more" you just load as many as you've counted. 

That is a hack to work around the challenge, but I think it might be the easiest short-term approach

Reply:
Okay! Can I do it in my next PR?

Reply:
The code is currently working such that if the user saves more than 10 places then only 10 cards will show but no errors